asdf/_lazy_nodes.py

```diff
@@ -0,0 +1,162 @@
+import collections
+import warnings
+from types import GeneratorType
+
+from . import tagged
+from .exceptions import AsdfConversionWarning
+from .extension._serialization_context import BlockAccess
+
+
+def _convert(value, af_ref):
+    if af_ref is None:
+        raise Exception("no ASDF for you!")
+    af = af_ref()
+    if af is None:
+        raise Exception("no ASDF for you!")
+    value_id = id(value)
+    if value_id in af._tagged_object_cache:
+        return af._tagged_object_cache[value_id][1]
+    extension_manager = af.extension_manager
+    sctx = af._create_serialization_context(BlockAccess.READ)
+    tag = value._tag
+    if not extension_manager.handles_tag(tag):
+        if not af._ignore_unrecognized_tag:
+            warnings.warn(
+                f"{tag} is not recognized, converting to raw Python data structure",
+                AsdfConversionWarning,
+            )
+        if isinstance(value, list):
+            obj = AsdfListNode(value, af_ref)
+        elif isinstance(value, dict):
+            obj = AsdfDictNode(value, af_ref)
+        else:
+            obj = value
+        af._tagged_object_cache[value_id] = (value, obj)
+        return obj
+    converter = extension_manager.get_converter_for_tag(tag)
+    data = value.data
+    if isinstance(data, dict):
+        data = AsdfDictNode(data, af_ref)
+    elif isinstance(data, list):
+        data = AsdfListNode(data, af_ref)
+    obj = converter.from_yaml_tree(data, tag, sctx)
+    if isinstance(obj, GeneratorType):
+        # TODO we can't quite do this for every instance
```

Comment:
astropy transform converters return `GeneratorType` (transforms can refer to their own inverse). This works for all transform tests but does not work for `FractionWithInverse`.

Reply:
This was resolved in d10e15ae12dd921c22f30fba65fadbf9efc20b94